Examples/ObjectiveC/CacheManagementExample.m

```diff
@@ -0,0 +1,149 @@
+    }];
+}
+
+// This method deletes the cache.db file, then reinitializes it. This deletes offline packs and ambient cache resources. You should not need to call this method. Invalidating the ambient cache and/or offline packs should be sufficient for most use cases.
+- (void)resetDatabase {
+    CFTimeInterval start = CACurrentMediaTime();
+    [[MGLOfflineStorage sharedOfflineStorage] resetDatabaseWithCompletionHandler:^(NSError * _Nullable error) {
+        if (error) {
+            NSLog(@"Error: %@", error.localizedDescription);
+            return;
+        }
+        CFTimeInterval difference = CACurrentMediaTime() - start;
+
+        // Display an alert to indicate that the cache.db file has been reset.
+        [self presentCompletionAlertWithTitle:@"Database reset" andMessage: [NSString stringWithFormat: @"Reset database in %f seconds", difference]];
+    }];
+}
+
+- (void)addOfflinePack {
+    MGLTilePyramidOfflineRegion *region = [[MGLTilePyramidOfflineRegion alloc] initWithStyleURL:self.mapView.styleURL bounds:self.mapView.visibleCoordinateBounds fromZoomLevel:0 toZoomLevel:2];
+
+    NSDictionary *info = @{@"name": @"Offline Pack"};
+    [NSKeyedArchiver archivedDataWithRootObject:info];
+
+    [[MGLOfflineStorage sharedOfflineStorage] addPackForRegion:region withContext:info completionHandler:^(MGLOfflinePack * _Nullable pack, NSError * _Nullable error) {
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+            return;
+        }
+
+        [pack resume];
+    }];
+
+}
+# pragma mark: Add UI components
+- (void)presentActionSheet {
+
+    if (!self.alertController) {
+        self.alertController = [UIAlertController alertControllerWithTitle:@"Cache Management Options" message:nil preferredStyle:UIAlertControllerStyleActionSheet];
+
+        [self.alertController addAction:[UIAlertAction actionWithTitle:@"Invalidate Ambient Cache" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+            [self invalidateAmbientCache];
+        }]];
+
+         [self.alertController addAction:[UIAlertAction actionWithTitle:@"Invalidate Offline Pack" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+            [self invalidateOfflinePack];
+        }]];
+
+          [self.alertController addAction:[UIAlertAction actionWithTitle:@"Clear Ambient Cache" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+             [self clearAmbientCache];
+         }]];
+
+           [self.alertController addAction:[UIAlertAction actionWithTitle:@"Reset Database" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+              [self resetDatabase];
+          }]];
+
+        [self.alertController addAction:[UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:nil]];
+    }
+
+    [self presentViewController:self.alertController animated:YES completion:nil];
```

Comment:
Check this functionality on iPad — action sheets aren’t modal on larger screen sizes, but instead are presented as popovers from a particular origin.

Reply:
Hm, so I'm running into an issue where the bar button appears on iPhones, but not on the iPad simulator.

Reply:
Okay, I think I've addressed this.